src/NumericalAlgorithms/Spectral/SwshDerivatives.cpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Spectral/SwshDerivatives.hpp"
+
+#include "DataStructures/ComplexDataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/ComplexDiagonalModalOperator.hpp"
+#include "DataStructures/ComplexModalVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"  // IWYU pragma: keep
+#include "DataStructures/TempBuffer.hpp"
+#include "NumericalAlgorithms/Spectral/ComplexDataView.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "NumericalAlgorithms/Spectral/SwshTransform.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/StaticCache.hpp"
+
+// IWYU pragma: no_forward_declare SpinWeighted
+
+namespace Spectral {
+namespace Swsh {
+namespace detail {
+
+template <typename DerivativeKind, int Spin>
```

Comment:
the caching functions should be in an anonymous namespace so they don't have external linkage